roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+import h5py
+from lazy_ops import DatasetView
+from ...segmentationextractor import SegmentationExtractor
+from ...extraction_tools import _pixel_mask_extractor
+import os
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._roi_response_fluorescence = self._roi_response
+        self._roi_response_neuropil = self._trace_extractor_read('C')
+        self._roi_response_deconvolved = self._trace_extractor_read('S')
+        self._images_correlation = self._summary_image_read()
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'][()]
+        self.image_masks = None
+
+    def __del__(self):
+        self._dataset_file.close()
+
+    def _file_extractor_read(self):
+        f = h5py.File(self.filepath, 'r')
+        return f
+
+    def _image_mask_sparse_read(self):
+        roi_ids = self._dataset_file['estimates']['A']['indices']
+        masks = self._dataset_file['estimates']['A']['data']
+        ids = self._dataset_file['estimates']['A']['indptr']
+        return masks, roi_ids, ids
+
+    def _trace_extractor_read(self, field):
+        extracted_signals = self._dataset_file['estimates'][field] # lazy read dataset)
+        return extracted_signals
+
+    def _summary_image_read(self):
+        if self._dataset_file['estimates'].get('Cn'):
+            summary_images_ = self._dataset_file['estimates']['Cn']
```

Comment:
why `summary_images_`, plural?

Reply:
yes your're right, its a bit misleading, i'll take a note and change this for all classes together in another PR

Reply:
naming have been fixed already